friendshipper/src-tauri/src/repo/operations/reset.rs

```diff
@@ -1,12 +1,214 @@
+    let uproject_path = PathBuf::from(app_config.repo_path).join(&uproject_path_relative);
+    let old_uproject: Option<UProject> = match UProject::load(&uproject_path) {
+        Err(e) => {
+            error!(
+                "Failed to load uproject before sync, skipping engine update. Error: {}",
+                e
+            );
+            None
+        }
+        Ok(uproject) => Some(uproject),
+    };
+
+    // update status
+    let aws_client: Option<AWSClient> = {
+        let client = ensure_aws_client(state.aws_client.read().await.clone())?;
+
+        // Make sure AWS credentials still valid
+        client.check_config().await?;
+        Some(client)
+    };
+    let storage: Option<ArtifactStorage> = {
+        match state.storage.read().clone() {
+            Some(storage) => Some(storage),
+            None => {
+                return Err(CoreError::Internal(anyhow!(
+                    "No storage configured for this app. AWS may still be initializing."
+                )))
+            }
+        }
+    };
+    let status_op = StatusOp {
+        repo_status: state.repo_status.clone(),
+        app_config: state.app_config.clone(),
+        repo_config: state.repo_config.clone(),
+        engine: state.engine.clone(),
+        git_client: state.git(),
+        github_username: state.github_username(),
+        aws_client,
+        storage,
+        allow_offline_communication: false,
+        skip_engine_update: false,
+    };
+    sequence.push(Box::new(status_op));
+
+    // download dlls
+    let aws_client = ensure_aws_client(state.aws_client.read().await.clone())?;
+    let project_name = RepoConfig::get_project_name(&state.repo_config.read().uproject_path)
+        .unwrap_or("unknown_project".to_string());
+    let artifact_prefix = state
+        .app_config
+        .read()
+        .clone()
+        .selected_artifact_project
+        .context("Project not configured. Repo may still be initializing.")?;
+    let uproject = UProject::load(&uproject_path)?;
+    let engine_path = state.app_config.read().get_engine_path(&uproject);
+
+    let download_dlls_op = DownloadDllsOp {
+        git_client: state.git(),
+        project_name,
+        dll_commit: state.repo_status.read().dll_commit_remote.clone(),
+        download_symbols: state.app_config.read().editor_download_symbols,
+        storage: state.storage.read().clone().unwrap(),
+        longtail: state.longtail.clone(),
+        tx: state.longtail_tx.clone(),
+        aws_client: aws_client.clone(),
+        artifact_prefix: artifact_prefix.clone(),
+        engine: state.engine.clone(),
+        engine_path,
+    };
+    sequence.push(Box::new(download_dlls_op));
+
+    // update engine
+    let app_config = state.app_config.read().clone();
+    let uproject_path =
+        PathBuf::from(&app_config.repo_path).join(&state.repo_config.read().uproject_path);
+    let new_uproject: Option<UProject> = match UProject::load(&uproject_path) {
```

Comment:
Since `sequence` is an async operation, this `new_project` will have the exact same contents as `old_project` because the repo hasn't been updated yet. I recommend putting all the logic that's in the handler here into `ResetToCommitOp::execute()` so all the operations happen as part of one "transaction" - this is how `PullOp` works right now.

Reply:
this has been moved into the :execute() and out of the handler